.travis_update_ghpages.sh

```diff
@@ -1,25 +1,43 @@
 #!/bin/bash
+
+set -o errexit -o nounset
+
+YELLOW=$'\e[1;33m'
+GREEN=$'\e[1;32m'
+RED=$'\e[1;31m'
+RESET=$'\e[0m'
+
 # Only publish from the main repository's master branch
 REPO_NAME="HydraCG/Specifications"
-if [ "$TRAVIS_REPO_SLUG" != "$REPO_NAME" ] || [ "$TRAVIS_BRANCH" != "master" ] || [ "$TRAVIS_PULL_REQUEST" != "false" ]; then exit; fi
-echo -e "Generating gh-pages...\n"
+if [ "$TRAVIS_REPO_SLUG" != "$REPO_NAME" ] || [ "$TRAVIS_BRANCH" != "master" ] || [ "$TRAVIS_PULL_REQUEST" != "false" ]; then
+  echo -e "${RED}This commit was made against $TRAVIS_BRANCH and not the master. Don't update gh-pages.${RESET}"
+  exit;
+fi
+
+echo -e "${YELLOW}Updating gh-pages...${RESET}"
+
+echo -e "${YELLOW}Generated files in $PWD/drafts/diagram${RESET}"
+ls drafts/diagram
 
 # Checkout the gh-pages branch
+echo -e "${YELLOW}Cloning and updating GitHub repository${RESET}"
 REPO_PATH=$PWD
 pushd $HOME
-git clone --quiet --branch=gh-pages https://${GH_TOKEN}@github.com/$REPO_NAME gh-pages 2>&1 > /dev/null
+git clone --branch=gh-pages https://${GH_TOKEN}@github.com/$REPO_NAME gh-pages
 cd gh-pages
-
-# Don't update if already at the latest version
-if [[ `git log -1 --pretty=%B` == *$TRAVIS_COMMIT* ]]; then exit; fi
+mkdir -p drafts/diagram/
 
 # Update documents
-cp -r $REPO_PATH .
+cp -rf $REPO_PATH/drafts/diagram/* drafts/diagram/
```

Comment:
Note: the reason I copied everything was to have a generic gh-page update script, which works for all future build steps we might need.